### PR TITLE
Upgrade packages in graphql pubspec.yaml and  graphql_flutter pubspec.yaml

### PR DESCRIPTION
**Pull request by hagen00**
Upgrade packages in `graphql` pubspec.yaml and `graphql_flutter` pubspec.yaml to latest, where possible.

### CODE REVIEW

**Comment by vincenzopalazzo** [comment]
```
Error: 84 tests passed, 2 failed.
--------------------------------------------------------------------------------

$ melos exec
  └> dart run test --coverage=coverage && dart run coverage:format_coverage --lcov --in=coverage --out=coverage.lcov --packages=.dart_tool/package_config.json --report-on=lib
     └> FAILED (in 1 packages)
        └> graphql (with exit code 1)

melos run client_test_coverage
  └> melos exec -- "dart run test --coverage="coverage" && dart run coverage:format_coverage --lcov --in=coverage --out=coverage.lcov --packages=.dart_tool/package_config.json --report-on=lib"
     └> FAILED
ScriptException: The script client_test_coverage failed to execute
make: *** [Makefile:38: ci_coverage_client] Error 1
```

probably there is some regression?

**Comment by hagen00** [comment]
If I remove the line `expect(socketClient.socketChannel!.closeCode, isNotNull);` in `disconnect via dispose` and `disconnect via dispose graphql-transport-ws` then all tests pass. 

I'm unable to figure out why `closeCode` is now null. There seem to be a number of issues related to `closeCode` on web_socket_channel's Github issue tracker.

**Comment by vincenzopalazzo** [comment]
Unfortunatly I have not enough bandwidth to debug this issue, I do not want to promise that I will review, it because I am not in a good period, sorry!

Probably if I were you I would upgrade one dependency at a time(and make a single commit per dependency and per graphql package) in this way you can isolate what is going wrong.

**Comment by hagen00** [comment]
I have run some tests and it seems the problem lies with web_socket_channel no longer setting closeCode. I was able to reproduce this easily with the same code working on version 2.4.5 and not working on 3.0.1

I have submitted an issue: https://github.com/dart-lang/web_socket_channel/issues/383

File: packages/graphql/test/websocket_test.dart [hunk 1]
```diff
@@ -178,24 +178,20 @@ Future<void> main() async {
       // The websocket should be in a fully closed state at this point,
       // we should have a confirmed close code in the channel.
       expect(socketClient.socketChannel, isNotNull);
-      expect(socketClient.socketChannel!.closeCode, isNotNull);
+      // as per https://github.com/dart-lang/web_socket_channel/issues/383, closeCode is not being sent in version 3.0.1
```
Comment (vincenzopalazzo) [inline packages/graphql/test/websocket_test.dart:181]:
```suggestion
      // FIXME: as per https://github.com/dart-lang/web_socket_channel/issues/383, closeCode is not being sent in version 3.0.1
```

**Review by vincenzopalazzo — COMMENTED** [review]
Make sense, but we just need to improve the git history sorry, because I need to have the ability to `git blame` or `git revert`.

So I suggest to make this PR in 2 different commit:

1. Upgrade the pubspec version and test in graphql package 
2. Updagade the pubspec version in graphql_flutter

Thanks!

**Comment by hagen00** [comment]
I have changed the dependency in `graphql_flutter` from `graphql: ^5.2.0-beta.8` to `graphql: ^5.2.0-beta.9`. I hope that was correct?

**Comment by vincenzopalazzo** [comment]
>I have changed the dependency in graphql_flutter from graphql: ^5.2.0-beta.8 to graphql: ^5.2.0-beta.9. I hope that was correct?

This should not be required, it is the dependencies resolution that will fix it for you, In this way you are forcing everyone to upgrade the packages, that can be a huge pain something

>The version in graphql is still version: 5.2.0-beta.9. I'm not sure if that also needs incrementing i.e to version: 5.2.0-beta.10.

package version are managed by the changelog tool, so leave them as it is.

This PR is missing just a big clean up inside the git history, 7 comments should be reduced in a smaller one.

A good reading https://cbea.ms/git-commit/

**Comment by hagen00** [comment]
@vincenzopalazzo I was trying to address this `2. Updagade the pubspec version in graphql_flutter`. I have now reverted it to be as before, i.e `graphql_flutter` is now once again `graphql: ^5.2.0-beta.8`

I'm not entirely sure what else I need to do. If there are mo changes required, please suggest them and I'll accept. 

I have also rebased into a single commit.

File: packages/graphql/pubspec.yaml [hunk 2]
```diff
@@ -32,4 +32,4 @@ dev_dependencies:
   lints: ^1.0.1
 
 environment:
-  sdk: '>=2.15.0 <4.0.0'
+  sdk: '>=3.3.0 <4.0.0'
```
Comment (vincenzopalazzo) [inline packages/graphql/pubspec.yaml:35]: why we need 3.3? and not just 3.0?

File: packages/graphql_flutter/pubspec.yaml [hunk 3]
```diff
@@ -12,9 +12,9 @@ dependencies:
   gql_exec: ^1.0.0
   flutter:
     sdk: flutter
-  meta: ^1.7.0
```
Comment (vincenzopalazzo) [inline packages/graphql_flutter/pubspec.yaml:15]: sorry this is painful I know, but this change need to go in a separate commit under the commit style `fix(graphql_flutter): upgrade meta,path,dart dependencies`

I am still not sure if the SDK version number is right, probably you can provide a justification for it?

File: packages/graphql/pubspec.yaml (hunk 2 above)
Comment (hagen00) [reply packages/graphql/pubspec.yaml]: I downgraded to 3.0.0 and ran `make check`. Tests still passed.

File: packages/graphql_flutter/pubspec.yaml (hunk 3 above)
Comment (hagen00) [reply packages/graphql_flutter/pubspec.yaml:15]: Thanks for your patience. Much appreciated. It's now a single commit (I did another rebase) with the message exactly `fix(graphql_flutter): upgrade meta,path,dart dependencies`. 

sdk requirements were also changed from 3.3 to 3.0. 

I'm not sure if you wanted two seperate PRs? Or whether it's ok now?

Comment (vincenzopalazzo) [reply packages/graphql_flutter/pubspec.yaml:15]: can you separate the change of `graphql` inside a commit and the change of `graphql_flutter` inside another one? I use an automate tool to do the release that it is based on this kind of rules.

I know it is annoying sorry!

Comment (hagen00) [reply packages/graphql_flutter/pubspec.yaml:15]: Ok, I have done this. I hope it's all done correctly. Please check again @vincenzopalazzo and merge if you are happy.

**Comment by hagen00** [comment]
@vincenzopalazzo is there a problem with this PR? Is the formatting incorrect? Must I redo things so that there are no formatting changes? Is everything fine otherwise i.e 2 commits, with the 2 commit messages?

**Comment by vincenzopalazzo** [comment]
Mh sorry for delay here

```
Run make ci_fmt_client
dart pub global run melos run client_analyze --no-select
melos run client_analyze
  └> melos exec -c 1 -- "dart format --set-exit-if-changed . && dart analyze . --fatal-infos"
     └> RUNNING

$ melos exec
  └> dart format --set-exit-if-changed . && dart analyze . --fatal-infos
     └> RUNNING (in 2 packages)

--------------------------------------------------------------------------------
example:
Formatted 3 files (0 changed) in 0.24 seconds.
Analyzing ....
No issues found!
example: SUCCESS
--------------------------------------------------------------------------------
graphql:
Formatted test/websocket_test.dart
Formatted 55 files (1 changed) in 0.[7](https://github.com/zino-hofmann/graphql-flutter/actions/runs/11727989636/job/32682149439?pr=1463#step:5:8)8 seconds.
--------------------------------------------------------------------------------

$ melos exec
  └> dart format --set-exit-if-changed . && dart analyze . --fatal-infos
     └> FAILED (in 1 packages)
        └> graphql (with exit code 1)

melos run client_analyze
  └> melos exec -c 1 -- "dart format --set-exit-if-changed . && dart analyze . --fatal-infos"
     └> FAILED
ScriptException: The script client_analyze failed to execute
make: *** [Makefile:32: ci_fmt_client] Error 1
```

**Comment by hagen00** [comment]
@vincenzopalazzo `melos run client_analyze` now succeeds and the commit history is still as it was before (2 commits, with the specific messages). 

Anything else that is required?

**Review by vincenzopalazzo — APPROVED** [review]
LGTM thanks! and sorry for delay
